pkg/scheduler/framework/types.go

```diff
@@ -213,8 +323,19 @@ func (ce ClusterEvent) IsWildCard() bool {
 // Note: we have a special case here when the coming event is a wildcard event,
 // it will force all Pods to move to activeQ/backoffQ,
 // but we take it as an unmatched event unless the ce is also a wildcard one.
-func (ce ClusterEvent) Match(event ClusterEvent) bool {
-	return ce.IsWildCard() || (ce.Resource == WildCard || ce.Resource == event.Resource) && ce.ActionType&event.ActionType != 0
+func (ce ClusterEvent) Match(incomingEvent ClusterEvent) bool {
+	return ce.IsWildCard() || ce.Resource.match(incomingEvent.Resource) && ce.ActionType&incomingEvent.ActionType != 0
+}
+
+// match returns true if the resource is matched with the coming resource.
+func (r EventResource) match(resource EventResource) bool {
+	// WildCard matches all resources
+	return r == WildCard ||
+		// Exact match
+		r == resource ||
+		// Pod matches assignedPod and unscheduledPod.
+		// (assignedPod and unscheduledPod aren't exposed and hence only used for incoming events and never used in EventsToRegister)
```

Comment:
so why are we making the distinction anyways? Is it just because of metrics?

Reply:
Yes, just because of metrics/logs that uses `event.Label()` fn.

So, instead of creating assignedPod/unscheduledPod resources, alternatively, we can utilize `label` field in `pkg/scheduler/framework/events.go` like -

```go
// Use `Pod` with the label
AssignedPodAdd = ClusterEvent{Resource: Pod, ActionType: Add, label: "AssignedPodAdd"}
// ... do the same with all the other AssignedPod/UnschedulablePod events.
```

I didn't go with that idea just because I wouldn't like to use `label` field as much as possible, and the current implementation allows us to remove `label` field eventually, after we remove `UnschedulableTimeout` (https://github.com/kubernetes/kubernetes/issues/87850).